src/m365/spo/commands/tenant/tenant-applicationcustomizer-set.ts

```diff
@@ -0,0 +1,169 @@
+    try {
+      const appCatalogUrl = await spo.getTenantAppCatalogUrl(logger, this.debug);
+      if (!appCatalogUrl) {
+        throw 'No app catalog URL found';
+      }
+
+      const listItem = await this.getListItem(logger, args.options, appCatalogUrl);
+      await this.updateTenantWideExtension(appCatalogUrl, args.options, logger, listItem);
+    }
+    catch (err: any) {
+      return this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getListItem(logger: Logger, options: Options, appCatalogUrl: string): Promise<ListItemInstance> {
+    const { title, id, clientSideComponentId } = options;
+    const filter = title ? `Title eq '${title}'` : id ? `Id eq '${id}'` : `TenantWideExtensionComponentId eq '${clientSideComponentId}'`;
+
+    if (this.verbose) {
+      logger.logToStderr(`Getting tenant-wide application customizer: "${title || id || clientSideComponentId}"...`);
+    }
+
+    const listServerRelativeUrl: string = urlUtil.getServerRelativePath(appCatalogUrl, '/lists/TenantWideExtensions');
+    const listItemInstances = await odata.getAllItems<ListItemInstance>(`${appCatalogUrl}/_api/web/GetList('${formatting.encodeQueryParameter(listServerRelativeUrl)}')/items?$filter=TenantWideExtensionLocation eq 'ClientSideExtension.ApplicationCustomizer' and ${filter}`);
+    if (listItemInstances) {
+      if (listItemInstances.length === 0) {
+        throw 'The specified application customizer was not found';
+      }
+
+      if (listItemInstances.length > 1) {
+        throw `Multiple application customizers with ${title ? `title '${title}'` : `ClientSideComponentId '${clientSideComponentId}'`} found. Please disambiguate using IDs: ${os.EOL}${listItemInstances.map(item => `- ${(item as any).Id}`).join(os.EOL)}`;
+      }
+    }
+    else {
+      throw 'The specified application customizer was not found';
+    }
+    return listItemInstances[0];
+  }
+
+  private async updateTenantWideExtension(appCatalogUrl: string, options: Options, logger: Logger, listItem: ListItemInstance): Promise<void> {
+    const { title, id, clientSideComponentId, newTitle, clientSideComponentProperties, webTemplate } = options;
+
+    if (this.verbose) {
+      logger.logToStderr(`Updating tenant-wide application customizer: "${title || id || clientSideComponentId}"...`);
+    }
+
+    const itemId = listItem.Id;
+    const commandOptions: spoListItemSetOptions = {
+      webUrl: appCatalogUrl,
+      listUrl: `${urlUtil.getServerRelativeSiteUrl(appCatalogUrl)}/Lists/TenantWideExtensions`,
+      id: itemId.toString(),
+      ...(newTitle && { Title: newTitle }),
+      ...(clientSideComponentProperties && { TenantWideExtensionComponentProperties: clientSideComponentProperties }),
+      ...(webTemplate && { TenantWideExtensionWebTemplate: webTemplate })
+    };
+
+    await Cli.executeCommandWithOutput(spoListItemSet as Command, { options: { ...commandOptions, _: [] } });
```

Comment:
We're moving away from using this way to call other commands. Instead we just want to call the rest API's and if necessary centralize that code. 

In this scenario: could you check how it's implemented on `tenant commandset set`?

https://github.com/pnp/cli-microsoft365/blob/main/src/m365/spo/commands/tenant/tenant-commandset-set.ts